city/crud.py

```diff
@@ -0,0 +1,54 @@
+from fastapi import HTTPException
+from sqlalchemy import insert
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.future import select
+
+from city import models, schemas
+
+
+async def get_all_cities(db: AsyncSession) -> list[models.City]:
+    query = select(models.City)
+    result = await db.execute(query)
+    return result.scalars().all()
+
+
+async def create_city(db: AsyncSession, city: schemas.CityCreate) -> dict:
+    query = insert(models.City).values(
+        name=city.name,
+        additional_info=city.additional_info,
+    ).returning(models.City.id)
+    result = await db.execute(query)
+    created_city_id = result.scalar_one()
+    await db.commit()
+    return {**city.model_dump(), "id": created_city_id}
```

Comment:
The method `model_dump()` is used here. Ensure that `schemas.CityCreate` has this method defined or imported correctly. If `model_dump()` is not available, consider using `city.dict()` to convert the Pydantic model to a dictionary.